readthedocs/donate/signals.py

```diff
@@ -1,9 +1,118 @@
+import random
+
 from django.dispatch import receiver
 from django.conf import settings
 from django.core.cache import cache
 
 from readthedocs.restapi.signals import footer_response
-from readthedocs.donate.models import SupporterPromo, VIEWS, CLICKS, OFFERS
+from readthedocs.donate.models import SupporterPromo, VIEWS, CLICKS, OFFERS, INCLUDE, EXCLUDE
+from readthedocs.donate.utils import get_ad_day
+
+
+PROMO_GEO_PATH = getattr(settings, 'PROMO_GEO_PATH', None)
+
+if PROMO_GEO_PATH:
+    import geoip2.database  # noqa
+    geo_reader = geoip2.database.Reader(PROMO_GEO_PATH)
+
+
+def show_to_geo(promo, country_code):
+    # Remove promo's that exclude this country.
+    for filter in promo.geo_filters.all():
+        if filter.filter_type == INCLUDE:
+            if country_code in filter.codes:
+                continue
+            else:
+                return False
+        if filter.filter_type == EXCLUDE:
+            if country_code in filter.codes:
+                return False
```

Comment:
Yeah, that's fine for now. Migrations could be difficult to reason about later, which is likely the only reason for thinking about it more early on. Maybe just have a ticket opened to track the change later